app/services/foreman_salt/fact_parser.rb

```diff
@@ -3,7 +3,8 @@ class FactParser < ::FactParser
     attr_reader :facts
 
     def operatingsystem
-      os = Operatingsystem.where(os_hash).first_or_initialize
+      os = Operatingsystem.new(os_hash)
+      os = Operatingsystem.find_by_attributes(**os_hash).first || os
```

Comment:
I'd like to avoid instantiating objects that may end up unused
```suggestion
      os = Operatingsystem.find_by_attributes(**os_hash).first || Operatingsystem.new(os_hash)
```